common/src/main/scala/org/apache/celeborn/common/rpc/netty/Inbox.scala

```diff
@@ -179,13 +198,12 @@ private[celeborn] class Inbox(
     }
   }
 
-  def post(message: InboxMessage): Unit = inbox.synchronized {
+  def post(message: InboxMessage): Unit = {
```

Comment:
Since `post` is not inside synchronize, it can happen that message be added into the queue after stop. How about the simple way that each time `post` is to be invoked, first check if size exceeds capacity and sleep a while if so?

Reply:
it is doable...

but I think given we already have built-in data structure to support "blocking when no capacity" with a simple `put()` call , we may not want to introduce a "not-that-elegant" way to handle this ?

regarding `it can happen that message be added into the queue after stop. `, I don't think it will happen IIUC?

inbox.stop() was called by dispatcher.stop() which was synchronized at dispatcher object, so we don't really need to synchronize here?

Reply:
You are right that message will not be added after stop is called, but seems `addMessage` can throw Exception if capacity is exceeded, is this what we want?

Reply:
If we use `put` here, we may face busy waiting if exceeds capacity inside the sync block. This is why I suggest to call `offer` and sleep a while outside the  sync block if false is returned.